playbooks/06-deploy-architecture.yml

```diff
@@ -27,7 +27,20 @@
       delegate_to: controller-0
       delegate_facts: true
 
-    - name: Create nova migration keypair
+    - name: Look for nova migration keypair file
+      register: _nova_key_file
+      ansible.builtin.stat:
+        path: "{{ cifmw_basedir }}/artifacts/nova_migration_key"
+
+    - name: Set _nova_key var when nova migration keypair exists
+      when: _nova_key_file.stat.exists
+      ansible.builtin.set_fact:
+        _nova_key:
+          filename: "{{ _nova_key_file.stat.path }}"
```

Comment:
you want to use the same "{{ _nova_key_file.stat.path }}" on line 47 so that we avoid repetition (I think this is what @lewisdenny said)